### DISCUSSION TITLE
Fully periodic Boundary Conditions

### DISCUSSION THREAD

**Discussion by RavisCPP** [idaholab/moose#28763] [Q&A Modules: Solid mechanics]
Hello everyone, 
I wish to apply fully periodic Boundary conditions in the solid mechanics module. I followed different discussions and came across this one discussion:
https://github.com/idaholab/moose/issues/22496

```
[BCs]
  [./Periodic]
    [./x_1]
      variable = disp_x
      primary = 'z0'
      secondary = 'z1'
      translation = '0 0 12'
    [../]
    [./x_2]
      variable = disp_x
      primary = 'y0'
      secondary = 'y1'
      translation = '0 12 0'
    [../]
    [./y_1]
      variable = disp_y
      primary = 'x0'
      secondary = 'x1'
      translation = '12 0 0'
    [../]
    [./y_2]
      variable = disp_y
      primary = 'z0'
      secondary = 'z1'
      translation = '0 0 12'
    [../]
    [./z_1]
      variable = disp_z
      primary = 'x0'
      secondary = 'x1'
      translation = '12 0 0'
    [../]
    [./z_2]
      variable = disp_z
      primary = 'y0'
      secondary = 'y1'
      translation = '0 12 0'
    [../]
  [../]
  [symmx] 
    type = DirichletBC
    variable = disp_x
    boundary = x0
    value = 0
  []

  [symmy] 
    type = DirichletBC
    variable = disp_y
    boundary = y0
    value = 0
  []

  [symmz] 
    type = DirichletBC
    variable = disp_z
    boundary = z0
    value = 0
  []

  [tdisp]
    type = FunctionDirichletBC
    variable = disp_z
    boundary = z1
    function = tdisp
  []
[]
```

The problem is this way I am able to apply BC in two directions, but not in the loading direction (Z-direction) because as soon as I apply periodic BC in Z direction. When using PBCs, the deformation on one boundary is matched by an equal and opposite deformation on the opposing boundary, leading to zero average stress, which is problematic in this case, where we are aiming for a non-zero stress state, especially in the loading direction.
Is there a way where we just apply the displacement in Z so that I can have periodicity around disp_z for Z face too?
Can anyone help me with that?

**Comment by GiudGiud** [comment]
Hello

> When using PBCs, the deformation on one boundary is matched by an equal and opposite deformation on the opposing boundary, leading to zero average stress,

Sure but that's what the periodic boundary condition means. If this is not the desired situation, then the system is simply not periodic. 

If you applied a load on the bottom towards the interior of the system, and expected a load on the top towards the interior of the system as well, then the system would not be periodic?

Guillaume

**Reply by RavisCPP** [reply]
I would like to clarify my question. I'm aiming for fully periodic boundary conditions while introducing a non-zero stress state. One approach that I tried was to apply the load to the middle elements of the mesh. 
This completely periodic BC is important for minimising surface effects in the simulations. I am looking for that kind of strategy, if possible. Thanks.

**Reply by GiudGiud** [reply]
@gambka do you know what they should do to achieve that?

**Reply by gambka** [reply]
I'm not sure. I'm still trying to wrap my head around the loading conditions. If possible, a diagram illustrating the desired BCs/loading conditions would be helpful for me to best provide guidance on how to achieve it in MOOSE.
